bin/node/runtime/pangolin/src/lib.rs

```diff
@@ -899,28 +902,92 @@ impl dvm_rpc_runtime_api::ConvertTransaction<OpaqueExtrinsic> for TransactionCon
 	}
 }
 
+fn migrate_treasury() {
+	// --- paritytech ---
+	use frame_support::{migration, StorageHasher, Twox64Concat};
+
+	type ProposalIndex = u32;
+
+	const OLD_PREFIX: &[u8] = b"DarwiniaTreasury";
+	const NEW_PREFIX: &[u8] = b"Treasury";
+
+	migration::move_storage_from_pallet(b"ProposalCount", OLD_PREFIX, NEW_PREFIX);
+	log::info!("`ProposalCount` Migrated");
+	migration::move_storage_from_pallet(b"Approvals", OLD_PREFIX, NEW_PREFIX);
+	log::info!("`Approvals` Migrated");
+
+	#[derive(Encode, Decode)]
+	struct OldProposal {
+		proposer: AccountId,
+		beneficiary: AccountId,
+		ring_value: Balance,
+		kton_value: Balance,
+		ring_bond: Balance,
+		kton_bond: Balance,
+	}
+	#[derive(Encode, Decode)]
+	struct Proposal {
+		proposer: AccountId,
+		value: Balance,
+		beneficiary: AccountId,
+		bond: Balance,
+	}
+	for (index, old_proposal) in migration::storage_key_iter::<
+		ProposalIndex,
+		OldProposal,
+		Twox64Concat,
+	>(OLD_PREFIX, b"Proposals")
+	.drain()
+	{
+		let hash = Twox64Concat::hash(&index.encode());
+
+		if old_proposal.ring_value != 0 {
+			let new_proposal = Proposal {
+				proposer: old_proposal.proposer.clone(),
+				value: old_proposal.ring_value,
+				beneficiary: old_proposal.beneficiary.clone(),
+				bond: old_proposal.ring_bond,
+			};
+
+			migration::put_storage_value(NEW_PREFIX, b"Proposals", &hash, new_proposal);
+		}
+		if old_proposal.kton_value != 0 {
+			let new_proposal = Proposal {
+				proposer: old_proposal.proposer,
+				value: old_proposal.kton_value,
+				beneficiary: old_proposal.beneficiary,
+				bond: old_proposal.kton_bond,
+			};
+
+			migration::put_storage_value(b"Instance2Treasury", b"Proposals", &hash, new_proposal);
+		}
+	}
+	migration::remove_storage_prefix(OLD_PREFIX, b"Proposals", &[]);
+	log::info!("`Proposals` Migrated");
+
+	migration::move_storage_from_pallet(b"Tips", OLD_PREFIX, NEW_PREFIX);
+	log::info!("`Tips` Migrated");
+	migration::move_storage_from_pallet(b"BountyCount", OLD_PREFIX, NEW_PREFIX);
```

Comment:
And the following bounty related storage.

Reply:
same question

Reply:
Same https://github.com/darwinia-network/substrate/blob/main/frame/bounties/src/lib.rs#L193